### DISCUSSION TITLE
Equivalent circuit parametrisation with DFN model

### DISCUSSION THREAD

**Discussion by Akila1993** [pybamm-team/PyBaMM#3823] [Q&A] [answered]
Hi everyone and @tinosulzer @brosaplanella @TomTranter ,

I am trying to treat Pybamm's DFN model as a real battery and want to identify the electrical parameters of the first-order equivalent circuit model based on EIS data. 

However I am not very clear how I could define the initial conditions of the ECM parameters (R1,R2,C1). The parameter set I am using for DFN model is Chen2020 and OKane2022. 

Can someone familiar with ECM advise me please. 

Thank you.

**Comment by TomTranter** [comment]
@Scottmar93 might have some experience of this

**Comment by Scottmar93** [comment, marked as answer]
Hi Akila, 

To get some initial estimates for the ECM parameters, I would recommend first plotting the EIS response of the DFN model on a nyquist plot. The series resistance is where the curve intersects the x-axis. The RC pair parameters  can be roughly estimated by looking at the semicircle: 
- the parallel resistance is the length along the x-axis of the semicircle 
- The capacitance is found by looking at the frequency, f, at the peak of the semicircle. The RC timescale, tau, is given by: tau=1/f. And this is related to the RC parameters by tau=R*C. So then you have C=tau/R. 

It’s not going to be easy to pick out the exact correct numbers but you should get some rough values that are good enough to start the optimization. You might also have more than one semicircle so you’ll need to choose the one you want to try to capture or add more RC pairs. 

A couple of points though: 
- the ECM structure that you are using won’t be able to capture the low frequency tail in the nyquist plot so if you feel you want to get better fits at low frequencies you might want to add a Warburg element in series. 
- I assume you are just using the simple formula Z=R1+ R2/(1+i*w*R2*C2), for doing the fitting which should be super fast. So you probably could just scan a bunch of different starting R1, R2, C1 parameters and see which one gives the best fit 
- if you want to use the ECM for simulating the cell during operation, I would probably recommend fitting out current pulses instead of EIS


I can recommend this site for a really good explanation on EIS 
https://lithiuminventory.com/experimental-electrochemistry/eis/rc-circuits/

**Reply by Akila1993** [reply]
Hi Scott, 

Thank you very much for your detailed reply. It clears me lot of doubts I had.

**Reply by DarioSlaifsteinSk** [reply]
I'm currently doing this for my own research.
If you are using the ECM for a dynamic simulations/control oriented applications check out the ESCtoolbox from Plett.
Here is a python implementation 
https://github.com/batterysim/esctoolbox-python

In my experience EIS fitting is highly manual and fragile (depends a lot on initial guess) to automate. This toolbox parametrizes OCV and all other parameters with linear algebra and (subspace) system identification so there's no risk of non-convergence.
